go-kit/metrics/provider/librato/librato.go

```diff
@@ -238,6 +262,19 @@ type attributes struct {
 	Aggregate bool `json:"aggregate,omitempty"`
 }
 
+// reportWithRetry the metrics to the url, every interval, with max retries.
+func (p *Provider) reportWithRetry(u *url.URL, interval time.Duration) error {
+	var err error
+
+	for i := p.numRetries; i > 0; i-- {
+		if err = p.report(u, interval); err == nil {
+			return nil
+		}
```

Comment:
If the err != nil we should probably call p.errorHandler as well and possibly modify the error to indicate if it's final or will be retried. Otherwise we only get a chance to notify the user of the library of the last error.